lib/profiler/get_tick.h

```diff
@@ -67,6 +67,19 @@ get_tick(void)
     return regs;
 }
 
+#elif defined(__APPLE__)
+#include <mach/mach_time.h>
+
+static __inline hrtime_t
+get_tick(void)
+{
+    static mach_timebase_info_data_t base;
+    uint64_t ticks;
+    (void) mach_timebase_info (&base);
+    ticks = mach_absolute_time ();
+    return (hrtime_t)((ticks * base.numer) / base.denom);
```

Comment:
How is Apple defining `hrtime_t`? Knowing that definition may help us avoid or polish this cast.

For example, some Linux-related [pages](https://www.unix.com/man-page/hpux/3C/gethrtime/) appears to define `hrtime_t` as `int64_t`.